File: 2023/day01/python/day01p1.py
#2023 Levi D. Smith
import sys

iSum = 0

for strLine in sys.stdin:
    strLine = strLine.rstrip()

    iFirst = None
    iLast = None

    iDigitStart = ord("0")
    iDigitEnd = ord("9")

    for c in strLine:
        if (ord(c) >= iDigitStart and ord(c) <= iDigitEnd):
            if (iFirst == None):
                iFirst = ord(c) - iDigitStart

            iLast = ord(c) - iDigitStart

    iValue = (iFirst * 10) + iLast
    iSum += iValue
print(iSum)

File: 2023/day01/ruby/day01p1.rb
#2023 Levi D. Smith

iSum = 0
while (strLine = gets()) 
  strLine.chomp!()

  iFirst = nil
  iLast = nil

  iDigitStart = "0".ord
  iDigitEnd = "9".ord
  strLine.each_byte do |c|
    if (c >= iDigitStart and c <= iDigitEnd)
      if (iFirst.nil?)
        iFirst = c - iDigitStart
      end
      iLast = c - iDigitStart
    end
  end

  iValue = (iFirst * 10) + iLast
#  puts "value: #{iValue}"
  iSum += iValue
end

puts iSum


